### ISSUE TITLE
Stemming (at least with Whoosh) should be disabled by default

### ISSUE THREAD

**Issue by rassie** [django-haystack/django-haystack#327]
Currently, Haystack's Whoosh backend automatically uses the StemmingAnalyzer, which leads to funny results, especially with people's names and in non-english environments. One example is the search term "Sodenthaler" (a water brand in Germany) which is reduced to Term('text', u'sodenthal') thinking it might be an adjective in comparative form. Therefore, the result of searching for "Sodenthaler" is under certain circumstances empty, while a search for "Sodenthal" yields something useful.

Long story short: the following lines in backends/whoosh_backend.py would fix that for now:

...
            else:
                analyzer = None
                if getattr(settings, 'HAYSTACK_WHOOSH_STEMMING', False):
                    analyzer = StemmingAnalyzer()
                schema_fields[field_class.index_fieldname] = TEXT(stored=True, analyzer=analyzer)
....

The else line is line number 146 in my current installation (should be 1.1.0)

**Comment by corro** [comment]
I am aware that this is an old issue, but I wanted you to know that this default behavior can indeed lead to strange search results. I did not expect this to be the default behavior and I even had to monkey patch the haystack whoosh backend in order to change this.

As a user, I would expect stemming to be disabled by default (like in whoosh itself) and to be configurable in the django settings (like @rassie proposed).

**Comment by denizdogan** [comment]
Note that if you pass `analyzer=None` to a field, Whoosh will default to using `StandardAnalyzer`, so the solution isn't as simple as described in the first post. (Maybe this has changed in the ten years that have passed since creating the issue, I just know what the current state of Whoosh is.) Additionally, if `analyzer` is None and `lang` is NOT None, it will default to using `LanguageAnalyzer(lang)`. Again, this is in Whoosh, not in Haystack.

I've created a PR for a related change, #1789, and if that one is approved and merged, my suggestion for this issue is:

Let the default be `TEXT(analyzer=None)` and let the users figure out what that means from reading the Whoosh documentation. It's the only thing that makes sense to me. Haystack adding its own defaults to Whoosh's defaults is confusing to everyone and causes more issues. This change is not backward compatible, so it would be best suited for inclusion in a major release.

Thoughts on the above?
